core/subsurface-qt/DiveObjectHelper.cpp

```diff
@@ -129,7 +129,7 @@ QString DiveObjectHelper::gps_decimal() const
 
 QVariant DiveObjectHelper::dive_site() const
 {
-	return QVariant::fromValue((void *)get_dive_site_by_uuid(m_dive->dive_site_uuid));
+	return QVariant::fromValue((uintptr_t)get_dive_site_by_uuid(m_dive->dive_site_uuid));
```

Comment:
I think I'm missing the significance of the `uintptr_t` here. Help?

Reply:
There's a comment in a later commit, which should probably be copied here. Short story: QVariant<void *> compare as equal in QML on my setup, whereas QVariant<uintptr_t> don't. I couldn't believe it either.